### PR TITLE
PIN-4054 - New DtD Catalog Exporter job with e-service total load

### PR DESCRIPTION
**Pull request by Carminepo2**
This job is not a cronjob, should only be started manually. 
The bucket output should be `interop-public-dashboards-${ENV}`.

### CODE REVIEW

File: jobs/dtd-catalog-total-load-exporter/metrics.json
Comment (galales) [inline jobs/dtd-catalog-total-load-exporter/metrics.json:1]: is the export intended to be committed?

File: jobs/dtd-catalog-total-load-exporter/result.json
Comment (galales) [inline jobs/dtd-catalog-total-load-exporter/result.json:1]: same for committed export

File: jobs/dtd-catalog-total-load-exporter/src/services/read-model-queries.service.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+              cond: {
+                $or: [{ $eq: ['$$descriptor.state', 'Published'] }, { $eq: ['$$descriptor.state', 'Suspended'] }],
+              },
+            },
+          },
+        },
+      },
+      {
+        $addFields: {
+          activeDescriptor: { $arrayElemAt: ['$activeDescriptor', 0] },
```
Comment (galales) [inline jobs/dtd-catalog-total-load-exporter/src/services/read-model-queries.service.ts]: we may have 
- descriptor1 Suspended
- descriptor2 Published
in this case the Suspended descriptor would be taken, is it ok?

File: jobs/dtd-catalog-total-load-exporter/package.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    "eslint": "8.46.0",
+    "prettier": "^3.0.1",
+    "ts-node": "^10.9.1",
+    "typescript": "5.1.6",
+    "vitest": "0.34.1"
+  },
+  "dependencies": {
+    "@aws-sdk/client-s3": "^3.387.0",
+    "axios": "^1.4.0",
+    "chalk": "^5.3.0",
```
Comment (galales) [inline jobs/dtd-catalog-total-load-exporter/package.json]: we should probably remove `chalk` from the template script 😄

File: jobs/dtd-catalog-total-load-exporter/metrics.json
Comment (Carminepo2) [reply jobs/dtd-catalog-total-load-exporter/metrics.json:1]: Oops, as always I forget to delete stuff before committing 🥲

File: jobs/dtd-catalog-total-load-exporter/package.json (hunk 2 above)
Comment (Carminepo2) [reply jobs/dtd-catalog-total-load-exporter/package.json]: Yup! Done!

File: jobs/dtd-catalog-total-load-exporter/src/services/read-model-queries.service.ts (hunk 1 above)
Comment (Carminepo2) [reply jobs/dtd-catalog-total-load-exporter/src/services/read-model-queries.service.ts]: It should prioritize the Published one, thanks for the heads up!

File: jobs/dtd-catalog-total-load-exporter/src/utils/helpers.utils.ts [hunk 3]
```diff
@@ -62,3 +62,21 @@ export function getAllAttributesIdsInEServicesActiveDescriptors(eservices: Array
 
   return Array.from(attributesIds)
 }
+
+export function getEServiceActiveDescriptor(eservice: EServiceQueryOutput): EServiceDescriptor {
```
Comment (galales) [inline jobs/dtd-catalog-total-load-exporter/src/utils/helpers.utils.ts]: this logic is duplicated [here](https://github.com/pagopa/interop-be-reports/pull/36/files/3e000db70e914e0b57ccf3e880ee4292f025271c..a47f1be13935a8557d7c308a84133f69d51c3bb4#diff-b2adce123984d3c8c0d83e443d51232849a5addc80d5e60a14599546a71be198R88).
I suggest to use just one reusable function

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**